src/pybind/mgr/dashboard/frontend/src/app/shared/components/copy2clipboard-button/copy2clipboard-button.component.ts

```diff
@@ -1,41 +1,30 @@
-import { Directive, ElementRef, HostListener, Input, OnInit, Renderer2 } from '@angular/core';
+import { Component, HostListener, Input } from '@angular/core';
 
 import { detect } from 'detect-browser';
 import { ToastrService } from 'ngx-toastr';
 
-@Directive({
-  selector: '[cdCopy2ClipboardButton]'
+@Component({
+  selector: 'cd-copy-2-clipboard-button',
+  templateUrl: './copy2clipboard-button.component.html',
+  styleUrls: ['./copy2clipboard-button.component.scss'],
 })
-export class Copy2ClipboardButtonDirective implements OnInit {
-  @Input()
-  private cdCopy2ClipboardButton: string;
+export class Copy2ClipboardButtonComponent {
   @Input()
+  private source: string;
   byId = true;
```

Comment:
Any specific reason of removing `@Input()` decorator?

Reply:
It looks like the diff didn't catch the fact there's a second `@Input()` implemented just below; I use `source` as the variable to hold the copied text; we drop the `OnInit` implementation since we do rendering in the component `html` file, rather than within the `ts` files. If you think there should be a second implementation of `Input()`, let me know, but I think it should be working as intended.

Reply:
Okay, if it's working then it's good to me to keep the change.